### PR TITLE
TW-323: RPC list altering update

### PR DESCRIPTION
**Pull request by alex-tsx**
- [tech] Fix postinstall script on Linux
- TW-323: RPC list altering update. 'Add node' -> '+' btn

### CODE REVIEW

File: src/components/styled-radio-group/radio-group/RadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useMemo } from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { RadioItem } from './RadioItem';
+import { RadioGroupProps } from './types';
+
+export function RadioGroup({ containerStyle, layout = 'column', onPress, items, selectedId }: RadioGroupProps) {
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/RadioGroup.tsx:7]: file name has wrong case (the same for RadioItem)

File: src/components/styled-radio-group/radio-group/RadioGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <View style={[styles.container, { flexDirection: layout }, containerStyle]}>
+      {itemsLocal.map(item => (
+        <RadioItem {...item} />
+      ))}
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/RadioGroup.tsx:40]: extract styles into separate file

File: src/components/styled-radio-group/radio-group/RadioGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          if (item.onPress) {
+            item.onPress(id);
+          }
+        }
+      })),
+    [items, selectedId]
+  );
+
+  return (
+    <View style={[styles.container, { flexDirection: layout }, containerStyle]}>
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/RadioGroup.tsx:32]: I think we will have 1 direction for now. You could just add default value into styles

File: src/components/styled-radio-group/radio-group/RadioItem.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  onPress,
+  selected = false,
+  size = 24
+}) => {
+  const borderWidth = PixelRatio.roundToNearestPixel(size * 0.1);
+  const sizeHalf = PixelRatio.roundToNearestPixel(size * 0.5);
+  const sizeFull = PixelRatio.roundToNearestPixel(size);
+
+  const orientation: ViewStyle = layout === 'column' ? { alignItems: 'center' } : { flexDirection: 'row' };
+  const margin: ViewStyle = layout === 'column' ? { marginTop: 10 } : { marginLeft: 10 };
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/RadioItem.tsx:32]: the same, you are overthinking about `layout`, we just don;t need that

File: src/components/styled-radio-group/radio-group/RadioItem.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  container: {
+    alignItems: 'center',
+    marginHorizontal: 10,
+    marginVertical: 5
+  },
+  border: {
+    justifyContent: 'center',
+    alignItems: 'center'
+  }
+});
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/RadioItem.tsx:92]: extract styles

File: src/components/styled-radio-group/radio-group/types.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  descriptionStyle?: object;
+  disabled?: boolean;
+  id: string;
+  label?: string;
+  labelStyle?: object;
+  layout?: 'row' | 'column';
+  onPress?: (id: string) => void;
+  size?: number;
+  value?: string;
+  buttons?: ItemButtonInterface[];
```
Comment (unixvb) [inline src/components/styled-radio-group/radio-group/types.ts:17]: remove what we don;t need

File: src/modals/custom-rpc-modals/add-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+export const AddCustomRpcModal: FC = () => {
+  const dispatch = useDispatch();
+  const { goBack } = useNavigation();
+  const rpcList = useRpcListSelector();
+
+  const handleSubmit = (newRpc: RpcInterface) => {
+    const duplicate = rpcList.find(rpc => rpc.name === newRpc.name || rpc.url === newRpc.url);
+
+    if (duplicate != null) {
```
Comment (unixvb) [inline src/modals/custom-rpc-modals/add-modal.tsx:34]: use `isDefined`

File: src/modals/custom-rpc-modals/edit-modal.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+    if (index < 0) {
+      return void showErrorToast({ description: `RPC not found ${initialValues.name}(${initialValues.url})` });
+    }
+
+    const otherItems = [...rpcList];
+    otherItems.splice(index, 1);
+    const duplicate = otherItems.find(rpc => rpc.name === values.name || rpc.url === values.url);
+
+    if (duplicate != null) {
```
Comment (unixvb) [inline src/modals/custom-rpc-modals/edit-modal.tsx:56]: the same code, mb create shared fn and reuse it?

File: src/store/settings/settings-reducers.ts [hunk 9]
```diff
@@ -128,3 +134,27 @@ export const settingsReducers = createReducer<SettingsState>(settingsInitialStat
     return state;
   });
 });
+
+const alterCustomRPC = (state: SettingsState, url: string, values?: RpcInterface) => {
```
Comment (unixvb) [inline src/store/settings/settings-reducers.ts:138]: mb move this fn out of `settings-reducers.ts` file?

File: src/modals/custom-rpc-modals/add-modal.tsx (hunk 7 above)
Comment (unixvb) [reply src/modals/custom-rpc-modals/add-modal.tsx]: it's very safe and comfortable that this function covers `undefined` and `null` cases and works as a type guard.  
you don't need to worry that something except "defined" type will pass trough

**Review by unixvb — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Review by NovakVlad2424 — APPROVED** [review]

**Pull request merged**